pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -135,12 +144,33 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
 	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers *cache.SocketPluginHandlers,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
 		client, conn, err := dial(socketPath, dialTimeoutDuration)
 		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+			var handlerWithName cache.NamedPluginHandler
+			handlerFound := false
+			pathToHandlers.Lock()
+			if len(pathToHandlers.Handlers) != 0 {
```

Comment:
This is not necessary here. You can directly do `handlerWithName, handlerFound = pathToHandlers.Handlers[socketPath]`.

Reply:
This check also covers the case where pathToHandlers.Handlers == nil.
I think we should keep the check.

Reply:
It's fine if pathToHandlers.Handlers is nil https://play.golang.org/p/MdgbAMEAdR6